### PR TITLE
L110: C# nullable reference types proposal

### PR DESCRIPTION
**Pull request by tonydnewell**
Proposal to support nullable reference types in the protocol buffers and gRPC plugin code generation.

### CODE REVIEW

File: L107-csharp-nullable-reference-types.md
Comment (jskeet) [inline L107-csharp-nullable-reference-types.md:38]: Mega nit: double space after "This"

File: L107-csharp-nullable-reference-types.md
Comment (jskeet) [inline L107-csharp-nullable-reference-types.md:40]: Possibly unwrap this, to be consistent with other paragraphs?

**Review by jskeet — APPROVED** [review]
Looks good. I think it would be worth mentioning the Google.Protobuf runtime, and how it doesn't currently have any nullability annotations. There's work ongoing to implement that, but it's very slow-moving (partly due to my lack of reviewing time).

**Review by JamesNK — APPROVED** [review]

File: L107-csharp-nullable-reference-types.md
Comment (jtattermusch) [inline L107-csharp-nullable-reference-types.md:145]: typo: "and when the enabled"

File: L107-csharp-nullable-reference-types.md
Comment (jtattermusch) [inline L107-csharp-nullable-reference-types.md:139]: I didn't know that we know so precisely what exactly what would change (sounded like the reality will be more complicated that this), but if we do, that's great. 
FWIW, I think we don't need to mention exactly what changes would be required here, as long as we describe the general idea behind the changes and approx. what they would be. So I think we could at least qualify the statements here to not be as prescriptive?

File: L107-csharp-nullable-reference-types.md
Comment (jtattermusch) [inline L107-csharp-nullable-reference-types.md:169]: nit: remove space before `]'

**Review by jtattermusch — COMMENTED** [review]
The proposal looks good to me (a few comments pending), but I'll leave it to @apolcyn, @jskeet  and @JamesNK  to actually LGTM.

@apolcyn will be listed as the Approver in the top of this grpc-proposal (as clearly proposals need to be approved by someone in the grpc team).

File: L107-csharp-nullable-reference-types.md
Comment (tonydnewell) [reply L107-csharp-nullable-reference-types.md:139]: I've now qualified this by saying that other changes may become apparent during development.

**Review by apolcyn — APPROVED** [review]

File: L107-csharp-nullable-reference-types.md
Comment (apolcyn) [inline L107-csharp-nullable-reference-types.md:1]: @tonydnewell can you please update this file name to `L110-csharp-nullable-reference-types.md`.

We can merge after that.

**Review by apolcyn — APPROVED** [review]

File: L107-csharp-nullable-reference-types.md
Comment (tonydnewell) [reply L107-csharp-nullable-reference-types.md]: @apolcyn renamed.

**Pull request merged**
